massa-execution/src/types.rs

```diff
@@ -374,20 +166,13 @@ pub(crate) enum ExecutionRequest {
     },
     /// Shutdown state, set by the worker to signal shutdown to the VM thread.
     Shutdown,
-    /// Get events by slot range
-    GetSCOutputEventBySlotRange {
-        start: Slot,
-        end: Slot,
-        response_tx: oneshot::Sender<Vec<SCOutputEvent>>,
-    },
-    /// Get events by caller
-    GetSCOutputEventByCaller {
-        caller_address: Address,
-        response_tx: oneshot::Sender<Vec<SCOutputEvent>>,
-    },
-    /// get events by smart contract
-    GetSCOutputEventBySCAddress {
-        sc_address: Address,
+    /// Get events with filter criteria
+    GetSCOutputEvents {
+        start_slot: Option<Slot>,
+        end_slot: Option<Slot>,
+        emitter_address: Option<Address>,
+        original_caller_address: Option<Address>,
+        origin_operation_id: Option<OperationId>,
         response_tx: oneshot::Sender<Vec<SCOutputEvent>>,
```

Comment:
Can we get some docs on each member please?

Reply:
I hopefully copy pasted the doc everywhere, but I'm not sure how clear it is.